src/VCR/Cassette.php

```diff
@@ -68,17 +68,45 @@ public function hasResponse(Request $request)
      */
     public function playback(Request $request)
     {
-        foreach ($this->storage as $recording) {
-            $storedRequest = Request::fromArray($recording['request']);
-            if ($storedRequest->matches($request, $this->getRequestMatchers())) {
-                return Response::fromArray($recording['response']);
+        if ($this->config->getPlaybackMode() === VCR::PLAYBACK_MODE_ALL_IN_ORDER) {
+            if ($this->storage->valid()) {
+                $recording = $this->storage->current();
+                $this->storage->next();
+
+                $storedRequest = Request::fromArray($recording['request']);
+                if ($storedRequest->matches($request, $this->getRequestMatchers())) {
+                    return Response::fromArray($recording['response']);
+                }
+            }
+        } else {
+            foreach ($this->storage as $recording) {
+                $storedRequest = Request::fromArray($recording['request']);
+                if ($storedRequest->matches($request, $this->getRequestMatchers())) {
+                    return Response::fromArray($recording['response']);
+                }
             }
```

Comment:
Looks like lines 76-79 are duplicated on lines 83-86. Can that bit be extracted to its own method?